scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -284,6 +288,34 @@ object Execution {
   }
 
   /**
+    * This is a Thread used as a shutdown hook to clean up temporary files created by some Execution
+    */
+  case class TempFileCleanup(filesToCleanup: mutable.Set[String], mode: Mode) extends Thread {
```

Comment:
since we're just iterating over the filesToCleanup in this case class, maybe we can make this a more generic interface rather than mutable.Set? Iterable / Seq / Set?